src/components/ChatRoomPage/MessageList.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect, useRef } from 'react';
+import profileIcon from "../../assets/ChatRoom/profile.svg";
+import { formatTime } from '../../utils/ClockUtils';
+
+interface User {
+    userId: number;
+    userName: string;
+}
+
+interface Message {
+    senderId: number;
+    text: string;
+    timestamp: Date;
+}
+
+interface MessageListProps {
+    messages: Message[];
+    currentUserId: number;
+    users: User[];
+}
+
+const groupMessages = (messages: Message[]) => {
+    return messages.reduce((arr, message, index) => {
+        const timeKey = formatTime(message.timestamp);
+        const prevMessage = messages[index - 1];
+        const isSameSender = prevMessage && prevMessage.senderId === message.senderId;
+        const isSameMinute = prevMessage && formatTime(prevMessage.timestamp) === timeKey;
+
+        if (!isSameSender || !isSameMinute) {
+            arr.push({ timeKey, senderId: message.senderId, messages: [message] });
+        } else {
+            arr[arr.length - 1].messages.push(message);
+        }
+
+        return arr;
+    }, [] as { timeKey: string; senderId: number; messages: Message[] }[]);
+};
+
+const MessageList: React.FC<MessageListProps> = ({ currentUserId, messages, users }) => {
+    const scrollRef = useRef<HTMLDivElement>(null);
+
+    // senderId로 사용자 이름 찾기
+    const getUserName = (senderId: number) => {
+        const user = users.find(user => user.userId === senderId);
+        return user ? user.userName : '(알 수 없음)';
+    };

```

Comment:
```suggestion
const getUserName = useCallback((senderId: number) => {
    const user = users.find(user => user.userId === senderId);
    return user ? user.userName : '(알 수 없음)';
}, [users]);

```
users가 변경되지 않으면 불필요하게 재생성될 필요가 없어서 useCallback을 사용하여 함수의 재생성을 방지하는 건 어떨까요?